public/locations-list-map.js

```diff
@@ -406,60 +483,85 @@ function centerMapToState(map, state) {
+  var latlng = new google.maps.LatLng(latitude, longitude);
 
-                markerDistances[distance] = marker;
+  //Compute the distances of all markers from the user
+  var markerDistances = new Map(); // an associative array containing the marker referenced by the computed distance
+  var distances = []; // all the distances, so we can sort and then call markerDistances
+  for (const marker of markers) {
+    let distance = google.maps.geometry.spherical.computeDistanceBetween(marker.position, latlng);
+    //console.log(marker);
 
-                distances.push(distance);
-            }
+    // HACK: In the unlikely event that the exact same distance is computed, add one meter to the distance to give it a unique distance
+    // This could occur if a marker was added twice to the same location.
+    if (markerDistances.has(distance)) { distance = distance + 1; }
 
-            // sort the distances and set bounds to closest three
-            distances.sort((a, b) => a - b);
+    markerDistances[distance] = marker;
 
+    distances.push(distance);
+  }
 
-            // center the map on the user
-            const bounds = new google.maps.LatLngBounds();
-            bounds.extend(user_latlng);
+  // sort the distances and set bounds to closest three
+  distances.sort((a, b) => a - b);
 
-            // Extend the bounds to contain the three closest markers
-            i = 0;
-            while (i < 3) {
-                // Get one of the closest markers
-                var distance = distances[i]
-                var marker = markerDistances[distance];
+  // center the map on the user
+  const bounds = new google.maps.LatLngBounds();
+  bounds.extend(latlng);
 
-                // Add to the iterator first just in case something fails later to avoid infinite loop
-                i++;
+  // Extend the bounds to contain the three closest markers
+  let i = 0;
+  while (i < 3) {
+    // Get one of the closest markers
+    let distance = distances[i];
+    let marker = markerDistances[distance];
 
-                const marker_lat = marker.position.lat();
-                const marker_lng = marker.position.lng();
+    // Add to the iterator first just in case something fails later to avoid infinite loop
+    i++;
 
-                const loc = new google.maps.LatLng(marker_lat, marker_lng);
-                bounds.extend(loc);
-            }
-            map.fitBounds(bounds);       // auto-zoom
-            map.panToBounds(bounds);     // auto-center
-        })
-    }
+    const marker_lat = marker.position.lat();
+    const marker_lng = marker.position.lng();
+
+    const loc = new google.maps.LatLng(marker_lat, marker_lng);
+    bounds.extend(loc);
+  }
+  map.fitBounds(bounds);       // auto-zoom
+  map.panToBounds(bounds);     // auto-center
+}
+
+
+
+/**
+ * Centers map at automatically detected coordinates using built in navigator.geolocation API.
+ */
+function centerMapToMarkersNearUser() {
```

Comment:
See, just moved down and pulled out contents and moved that to it’s own new reusable function.